chain/rpc/src/rpc.rs

```diff
@@ -498,6 +504,49 @@ mod tests {
         Ok(())
     }
 
+    #[async_std::test]
+    async fn test_should_estimate_tx() -> anyhow::Result<()> {
+        let _ = env_logger::builder().is_test(true).try_init();
+
+        let expected_block_time = Duration::from_secs(1);
+        let anvil = hopr_chain_types::utils::create_anvil(Some(expected_block_time));
+        let chain_key_0 = ChainKeypair::from_secret(anvil.keys()[0].to_bytes().as_ref())?;
+
+        let cfg = RpcOperationsConfig {
+            chain_id: anvil.chain_id(),
+            tx_polling_interval: Duration::from_millis(10),
+            expected_block_time,
+            finality: 2,
+            ..RpcOperationsConfig::default()
+        };
+
+        let client = JsonRpcProviderClient::new(
```

Comment:
This test now depends on external comms right?
It should be mocking the HttpRequestor trait to rather simulate the traffic.